waku-bindings/src/node/filter.rs

```diff
@@ -38,31 +47,35 @@ pub fn waku_filter_subscribe(
             &mut closure as *mut _ as *mut c_void,
         );
 
-        drop(CString::from_raw(pubsub_topic_ptr));
-        drop(CString::from_raw(content_topics_ptr));
-
         out
     };
 
+    notify.notified().await; // Wait until a result is received
     handle_no_response(code, result)
 }
 
-pub fn waku_filter_unsubscribe(
+pub async fn waku_filter_unsubscribe(
     ctx: &WakuNodeContext,
     pubsub_topic: &PubsubTopic,
     content_topics: Vec<WakuContentTopic>, // comma-separated list of content topics
 ) -> Result<()> {
     let content_topics_topics = WakuContentTopic::join_content_topics(content_topics);
 
-    let pubsub_topic_ptr = CString::new(String::from(pubsub_topic))
-        .expect("CString should build properly from pubsub topic")
-        .into_raw();
-    let content_topics_topics_ptr = CString::new(content_topics_topics)
-        .expect("CString should build properly from content topic")
-        .into_raw();
+    let pubsub_topic = CString::new(String::from(pubsub_topic))
+        .expect("CString should build properly from pubsub topic");
+    let pubsub_topic_ptr = pubsub_topic.as_ptr();
 
-    let mut result: LibwakuResponse = Default::default();
-    let result_cb = |r: LibwakuResponse| result = r;
+    let content_topics_topics = CString::new(content_topics_topics)
+        .expect("CString should build properly from content topic");
+    let content_topics_topics_ptr = content_topics_topics.as_ptr();
+
+    let mut result = LibwakuResponse::default();
+    let notify = Arc::new(Notify::new());
+    let notify_clone = notify.clone();
+    let result_cb = |r: LibwakuResponse| {
+        result = r;
+        notify_clone.notify_one(); // Notify that the value has been updated
+    };
```

Comment:
I think this pattern its ok but should be simpler if you use a channel to return the result. You can probably use [`kanal`](https://crates.io/crates/kanal) a mixed flavored channel (sync/async). To simplify it.